src/Common/src/CoreLib/System/MemoryExtensions.cs

```diff
@@ -245,12 +251,34 @@ public static bool SequenceEqual<T>(this Span<T> first, ReadOnlySpan<T> second)
             where T : IEquatable<T>
         {
             int length = first.Length;
-            if (typeof(T) == typeof(byte))
+            if (typeof(T) == typeof(byte) || typeof(T) == typeof(sbyte))
                 return length == second.Length &&
                 SpanHelpers.SequenceEqual(
                     ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(first)),
                     ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(second)),
                     length);
+
+            if (typeof(T) == typeof(char) || typeof(T) == typeof(short) || typeof(T) == typeof(ushort))
```

Comment:
I would imagine we could do similar optimizations for other methods that specialize for T == byte.
Should we extract this type of logic out into helpers methods now (or do that later)?

Reply:
We could do that later. I wasn't sure if the JIT could remove the overhead of the extra call if we did the type dispatching inside the helper. If it can, then yes, it makes more sense for the generic SequentialEqual helper to farm out to the optimized subversions.